src/app/[locale]/incident/add/components/questions/AssetSelect.tsx

```diff
@@ -1,8 +1,149 @@
-import { QuestionField } from '@/types/form'
-import { Paragraph } from '@/components/index'
+import { LocationMap } from '@/components/ui/LocationMap'
+import { MapDialog } from '@/app/[locale]/incident/add/components/MapDialog'
+import { PublicQuestion } from '@/types/form'
+import { MapProvider, MapRef, useMap } from 'react-map-gl/maplibre'
+import { useFormContext } from 'react-hook-form'
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Paragraph,
+  LinkButton,
+  Fieldset,
+  FieldsetLegend,
+} from '@/components/index'
+import { useFormStore } from '@/store/form_store'
+import { getNearestAddressByCoordinate } from '@/services/location/address'
+import { useConfig } from '@/hooks/useConfig'
+import { isCoordinates } from '@/lib/utils/map'
+import { useTranslations } from 'next-intl'
+import { FormFieldErrorMessage } from '@/components'
+import { getGeoJsonFeatures } from '@/services/location/features'
+import { FeatureCollection } from 'geojson'
 
-interface AssetSelectProps extends Omit<QuestionField, 'register' | 'errors'> {}
+export interface AssetSelectProps {
+  field?: PublicQuestion
+}
 
 export const AssetSelect = ({ field }: AssetSelectProps) => {
-  return <Paragraph>{field.key}</Paragraph>
+  const {
+    formState: { errors },
+  } = useFormContext()
+  const errorMessage = errors['location']?.message as string
+  const { formState: formStoreState } = useFormStore()
+  const [address, setAddress] = useState<string | null>(null)
+  const { config } = useConfig()
+  const t = useTranslations('describe-add.map')
+  const [dialogMap, setDialogMap] = useState<MapRef | null>(null)
+  const [features, setFeatures] = useState<FeatureCollection | null>(null)
+  const { formState } = useFormStore()
+
+  useEffect(() => {
+    const getAddress = async () => {
```

Comment:
de getAddress hier en in locationSelect is nu hetzelfde, die zouden we eventueel op een gedeelde plek kunnen zetten